fda/extrapolation.py

```diff
@@ -0,0 +1,332 @@
+            eval_points = eval_points.reshape(eval_points.shape[0])
+
+
+        return fdata(eval_points, derivative=derivative,
+                     extrapolation=ExtrapolationType.NONE, keepdims=keepdims)
+
+
+class ExceptionExtrapolation(Extrapolator):
+    """Raise and exception if a point is evaluated outside the domain range."""
+
+    def extrapolate(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        if eval_points.ndim == 3:
+            n_points = eval_points.shape[1]
+        else:
+            n_points = eval_points.shape[0]
+
+        raise ValueError(f"Attempt to evaluate {n_points} points outside the "
+                         f"domain range.")
+
+class FillExtrapolation(Extrapolator):
+    """The values outside the domain range will be filled with a fixed value."""
+
+    def __init__(self, fill_value):
+        self.fill_value = fill_value
+
+        super().__init__()
+
+    def extrapolate(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        # Case all the samples evaluates at same time
```

Comment:
Wow. Maybe I have misunderstood what this method should do, but I though that it should only return `np.full` with the same shape as the points to be evaluated. I do not understand what it actually does.